text/0000-new-test-waiters.md

```diff
@@ -0,0 +1,111 @@
+
+To address this, a new addon was written to experiment on a new test waiter system that would provide a number of things (as noted above):
+
+1. A new API that's _explicit_ and _straightforward_
+1. A more robust way to gather debugging information for the test waiter
+1. Default test waiters with the ability to author your own, more complex test waiters
+
+This would allow developers to utilize this new test waiters system to annotate their asynchronous operations not tracked by the settled check, and for those annotations to provide useful debugging information in the event their async extended past the expected duration of the test.
+
+## Comparison of old waiters system to new
+
+In the old test waiters system, you would do the following:
+
+```js
+import { registerWaiter } from '@ember/test';
+
+registerWaiter(function() {
+  return myPendingTransactions() === 0;
+});
+```
+
+While reading the above is straightforward, when writing a test waiter using the old system it's easy to forget what the expected return value is: `true` or `false`. Additionally, it's a bit more cognitive overhead to derive what the intended result of the particular boolean return value is: does returning `true` result in the test waiter waiting or not?
+
+As mentioned before, there's no additional information provided via `registerWaiter`, and no way to capture stack traces at the call site.
+
+The new test waiters system looks like this:
+
+```js
+import Component from '@ember/component';
+import { buildWaiter } from 'ember-test-waiters';
+
+let waiter = buildWaiter('friend-waiter');
+
+export default class Friendz extends Component {
+  didInsertElement() {
+    let token = waiter.beginAsync();
+
+    someAsyncWork()
+      .then(() => {
+        //... some work
+      })
+      .finally(() => {
+        waiter.endAsync(token);
+      });
+  }
+}
+```
+
+In the above example, a new test waiter is built that is identified via a `name` string passed into the `buildWaiter` function. This allows the waiter to be identifiable, and that name is ultimately used with test isolation validation to help developers narrow down problems in their tests.
+
+The API used to signal whether an asynchronous operation has begun and ultimately ended is through the paired calls of `beginAsync` and `endAsync`: begin to denote the start of the asynchronous operation, and end to denote the end. Unique instances of async operations are identified using a `token` returned from `beginAsync`, which is subsequently provided to the `endAsync` call.
+
+This new test waiters system has been through multiple iterations of refinement, and is in use and integrated with the test isolation validation system. Furthermore, it is backwards compatible with the old test waiters system, allowing applications and addons to gradually migrate to using the new system.
+
+## Deprecating the old test waiters
+
+The old test waiters system will be deprecated in its own deprecation RFC.
```

Comment:
Let's tweak this to be an overall "compatibility" section with:

* registerWaiter continues to work, and is not deprecated
* addons using ember-test-waiters work even if consumed in applications that do *not* use a new enough ember-test-helpers version (they don't get the nice new fancy helpful output, but it still works)